File: Accenture/6.ModifySentence/code.cpp
#include <iostream>
#include <sstream>
#include <vector>
using namespace std;
string shorten_word(const string& word) {
    if (word.length() > 10) {
        return word[0] + to_string(word.length() - 2) + word.back();
    }
    return word;}
int main() {
    string sentence, word;
    getline(cin, sentence);

    stringstream ss(sentence);
    vector<string> words;
    while (ss >> word) {
        words.push_back(shorten_word(word));
    }
    for (size_t i = 0; i < words.size(); i++) {
        cout << words[i];
        if (i != words.size() - 1) {
            cout << " ";
        }}
    cout << endl;
    return 0;}

File: Accenture/6.ModifySentence/code.py
def shorten_word(word):
    if len(word) > 10:
        return word[0] + str(len(word) - 2) + word[-1]
    return word

def shorten_sentence(sentence):
    words = sentence.split()
    return ' '.join([shorten_word(word) for word in words])

# Test case
sentence = "The demonstration of governments laws makes it difficult to understanding"
print(shorten_sentence(sentence))  # Output: "the d11n of g9s laws makes it difficult to u11g"
